fab/src/main/java/uk/co/markormesher/android_fab/FloatingActionButton.java

```diff
@@ -147,7 +150,10 @@ private void resetIcon() {
 	 */
 	public void setBackgroundColour(int colour) {
 		savedBgColour = colour;
-		cardView.setCardBackgroundColor(colour);
+		final Drawable background = fab.getBackground();
+		if (background == null) return;
+
```

Comment:
This needs to report a failure if the change was aborted here. The `background` variable is also unnecessary - it's accessed just once, so you can put `fab.get...` in the `if` condition.